src/psyclone/psyir/nodes/array_mixin.py

```diff
@@ -320,6 +318,30 @@ def indices(self):
                     f"expression but found '{type(child).__name__}'")
         return self.children
 
+    @property
+    def shape(self):
```

Comment:
I take your point here. However, it causes problems because I use it twice in `StructureReference.datatype`. One of these is because I'm also supporting `ArrayOfStructuresReference` with that method and the other is needed as I walk down the members of the structure access.

Reply:
I've kept it as a 'private' method and disable the pylint warning at the two locations. See what you think.